src/test/java/com/sleekbyte/tailor/functional/yaml/YamlConfigurationTest.java

```diff
@@ -0,0 +1,353 @@
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_1);
+
+        // Create config file that wants to only analyze YAML_TEST_1.swift
+        File configurationFile = includeOptionConfig(".tailor.yml");
+        String[] command = new String[] {
+            "--config", configurationFile.getAbsolutePath(),
+            "--no-color"
+        };
+        runTest(command);
+    }
+
+    @Test
+    public void testExceptOption() throws IOException {
+        // Add expected output
+        addExpectedMsg(8, 33,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        addExpectedMsg(9, 16,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        addExpectedMsg(10, 23,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        addExpectedMsg(11, 39,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        addExpectedMsg(12, 2,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        addExpectedMsg(14, 23,
+            Rules.TRAILING_CLOSURE,
+            Messages.CLOSURE + Messages.TRAILING_CLOSURE,
+            YAML_TEST_2);
+
+        addExpectedMsg(18, 2,
+            Rules.TERMINATING_SEMICOLON,
+            Messages.STATEMENTS + Messages.SEMICOLON,
+            YAML_TEST_2);
+
+        // Create config file that will only analyze YAML_TEST_2.swift for violations other than upper-camel-case
+        File configurationFile = exceptOptionConfig(".tailor.yml");
+        String[] command = new String[] {
+            "--config", configurationFile.getAbsolutePath(),
+            "--no-color"
+        };
+        runTest(command);
+    }
+
+    @Test
+    public void testOnlyExceptOptionPrecedence() throws IOException {
+        // Add expected output
+        addExpectedMsg(3, 7,
+            Rules.UPPER_CAMEL_CASE,
+            Messages.CLASS + Messages.NAMES + Messages.UPPER_CAMEL_CASE,
+            YAML_TEST_1);
+
+        addExpectedMsg(7, 7,
+            Rules.UPPER_CAMEL_CASE,
+            Messages.CLASS + Messages.NAMES + Messages.UPPER_CAMEL_CASE,
+            YAML_TEST_1);
+
+        // Create config file that wants to only analyze YAML_TEST_1.swift for upper-camel-case violations
+        // Test for precedence of `only` over `except`
+        File configurationFile = onlyAndExceptPrecedenceConfig(".tailor.yml");
+        String[] command = new String[] {
+            "--config", configurationFile.getAbsolutePath(),
+            "--no-color"
+        };
+        runTest(command);
+    }
+
+    @Test
+    public void testCliConfigOnlyOptionPrecedence() throws IOException {
```

Comment:
Can we change the name to `testCliAndConfigFilePrecendence` or is there a reason we put `..OnlyOptionPrecedence` in the name?